src/t_pstring.c

```diff
@@ -738,6 +738,7 @@ int pmSetCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
 int setRangeFunc(const char *old_val, size_t old_val_len, char **new_val,
                  size_t *new_val_len, void *args_pointer) {
   // assert(args_pointer);
+  *new_val = (char *)malloc(MAX_KVDK_STRING_SIZE);
```

Comment:
malloc MAX_KVDK_STRING_SIZE at line 741 may trigger

memory leak at line 768
memory space waste if each time allocate a MAX_KVDK_STRING_SIZE space for a small string
if we just want to fix max length case for:
pm.setrange mykey 536870910 world

can we consider add something like if(offset+val_len > MAX_KVDK_STRING_SIZE) {return ERR; } around line 768?